src/open_inwoner/cms/plugins/cms_plugins/tasks.py

```diff
@@ -88,19 +117,45 @@ def render(self, context, instance, placeholder) -> dict[str, str]:
 
         return context
 
+    def fetch_zgw_taken(self, user: User) -> list[OpenstaandeTaak]:
+        """
+        Fetch `openstaande taken` from ZGW API's
+        """
+        if not user.bsn:
+            return []
```

Comment:
Reading this again, this feels like a bit of a smell to me: if a `bsn` is necessary for this call to work, it should be an argument, and the caller should do this check.


```suggestion
    def fetch_zgw_taken(self, bsn: str) -> list[OpenstaandeTaak]:
        """
        Fetch `openstaande taken` from ZGW API's
        """
```

Reply:
The check is actually already done by the calling function, should have refactored this, thanks